addons/project-source-button/viewer.js

```diff
@@ -0,0 +1,171 @@
+const queries = (() => {
+  const query = window.location.search.slice(1);
+  const queries = {};
+  if (!query) return {};
+  for (let text of query.split("&")) {
+    text = text.split("=");
+    queries[text[0]] = text[1];
+  }
+  return queries;
+})();
+async function loadDom() {
+  window.addEventListener("load", (e) => {
+    Promise.resolve();
+  });
+}
```

Comment:
I don't think you quite understand Promises. `Promise.resolve()` does nothing here, since it just returns a promise that resolves immediately. Try this instead:

```js
function loadDom() {
  return new Promise((resolve) => {
    window.addEventListener("load", () => resolve());
  });
}
```

Reply:
I fixed it.